loaders/src/glTF/2.0/Extensions/EXT_mesh_gpu_instancing.ts

```diff
@@ -83,28 +98,60 @@ export class EXT_mesh_gpu_instancing implements IGLTFLoaderExtension {
             loadAttribute("SCALE");
 
             if (instanceCount == 0) {
+                for (const babylonMesh of node._primitiveBabylonMeshes) {
+                    if ((babylonMesh as any).thinInstanceSetBuffer) {
+                        (babylonMesh as Mesh)._thinInstanceDataStorage.instancesCount = 0;
```

Comment:
why do we need to use private member?

Reply:
Because setting the `thinInstanceCount` property won't work (that's what I did first) because there's no thin instance matrix buffer yet, so you can't set a `thinInstanceCount` property greater than 0.

This is a kind of hack to make sure that `hasThinInstances` will return `true` in line #784 of **glTFLoader.ts**, because this line can be executed before we get to the point where we set the matrix buffer in EXT_mesh_gpu_instancing class. At which point `thinInstanceCount` will be set to the right value.

Reply:
maybe we can add a prepareThinInstanceBuffer() or something

Reply:
So, that would be something like `thinInstancePrepareBuffer(kind: string, numInstances: number = 32, stride: number = 16)` that would allocate a matrix buffer big enough for the number of instances passed in parameter? Or should it be used only to pre-allocate matrix storage (so we could get rid of kind and stride)?

Reply:
We may not need this if I can remove all my ugly code and disable `InstancedMesh` optimization when the gpu_instancing extension is used. Waiting for @bghgary answers.